src/modules/RootPart.rb

```diff
@@ -973,222 +880,100 @@ def MountFSTab(fstab, _message)
       success
     end
 
-    # Mount /var partition
+    # Displays a warning dialog to the suer when mount failed
     #
-    # @param device [String] name of the device holding /var
-    # @return [String, nil] nil on success, error description on fail
-    def MountVarPartition(device)
-      mount_err = FsckAndMount("/var", device, "")
-
-      return nil unless mount_err
-
-      log.error("failed to mount /var")
-
-      # TRANSLATORS: error message when /var partition cannot be mounted. %{device}
-      # is replaced by a device name (e.g., /dev/sda2) and %{error} is replaced by
-      # error details.
-      format(
-        _("The /var partition %{device} could not be mounted.\n\n%{error}"),
-        device: device,
-        error:  mount_err
-      )
-    end
-
-    # <-- BNC #448577, Cannot find /var partition automatically
-    # returns if successful
-    def MountUserDefinedVarPartition
-      # function return value
-      manual_mount_successful = false
-
-      list_of_devices = []
-      # $[ "/dev/sda3" : "Label: My_Partition" ]
-      device_info = {}
-
-      # Creating the list of known partitions
-      Builtins.foreach(Storage.GetOndiskTarget) do |_device, description|
-        Builtins.foreach(Ops.get_list(description, "partitions", [])) do |partition|
-          # Some partitions logically can't be used for /var
-          next if Ops.get_symbol(partition, "detected_fs", :unknown) == :swap
-          next if Ops.get_symbol(partition, "type", :unknown) == :extended
-          next if !Builtins.haskey(partition, "device")
-
-          list_of_devices = Builtins.add(
-            list_of_devices,
-            Ops.get_string(partition, "device", "")
-          )
-          Ops.set(
-            device_info,
-            Ops.get_string(partition, "device", ""),
+    # Apart from informing, it lets on the user the next actions: to ignore the error and continue,
+    # to check and/or specify the mount options, or to abort the update.
+    #
+    # FIXME: this dialog should live in its own class. However, extracting it to a method
+    # looks like a good compromise in the context of the PBI it has been addressed. So please,
+    # feel free to make it a first-class citizen in future changes.
+    #
+    # @param spec [String]
+    # @param error [String] the error returned by {#FsckAndMount}
+    # @return [Symbol] the action chosen by the user, namely
+    #   :cont if decides to continue because the partition is not necessary for the update
+    #   :cmd when wants to check or specify the mount options
+    #   :cancel whether goes for aborting the update process
+    def mount_faild_action(spec, error)
```

Comment:
Why "faild" instead of "failed"?

Reply:
Because I'm starving and I have eaten the "e"? :stuck_out_tongue_winking_eye: Fixing it, thanks for catching.